src/checker.ml

```diff
@@ -872,12 +873,14 @@ let entrypoint_touch (state, _: checker * unit) : (LigoOp.operation list * check
 (**                               ORACLE                                     *)
 (* ************************************************************************* *)
 
-let entrypoint_receive_price (state, idx: checker * Ligo.nat) : (LigoOp.operation list * checker) =
+let entrypoint_receive_price (state, oracle_price: checker * (Ligo.nat * Ligo.nat)) : (LigoOp.operation list * checker) =
   assert_checker_invariants state;
   if !Ligo.Tezos.sender <> state.external_contracts.oracle then
     (Ligo.failwith error_UnauthorisedCaller : LigoOp.operation list * checker)
   else
-    let price = fixedpoint_of_ratio_floor (make_ratio (Ligo.int idx) tok_scaling_factor_int) in
+    let price =
+      let (num, den) = oracle_price in
+      fixedpoint_of_ratio_floor (make_ratio (Ligo.int num) (Ligo.int den)) in
```

Comment:
Maybe we should add a comment here to make it extra obvious that we are losing some precision?

Reply:
Ah yes, I added a small note in a9d6a88 :+1: